### PR TITLE
Align the Popup Menu under its child

### LINKED ISSUE #125474: `PopupMenuButton` - `padding` shouldn't be used when `PopupMenuPosition.under` is used with a child
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Run `flutter run` on the code example given (see below)
2. Notice that the menu has an offset of 4px

### Expected results

The menu should be aligned with the bottom of the child.

`padding` is only used by the `IconButton` when `child` is `null`. When `child` is not `null`, `padding` is never used in the `build`, so it shouldn't change the position of the menu.

### Actual results

The menu is overlapping the child by 4 px.

### Code sample

<details><summary>Code sample</summary>

You can also checkout https://github.com/ValentinVignal/flutter_app_stable/tree/flutter/padding-popup-menu-button

```dart
// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: true),
      home: Material(
        child: Column(
          children: <Widget>[
            Center(
              child: PopupMenuButton<void>(
                position: PopupMenuPosition.under,
                itemBuilder: (BuildContext context) {
                  return <PopupMenuEntry<Object>>[
                    const PopupMenuItem<Object>(
                      enabled: false,
                      child: Text('disabled'),
                    ),
                    const PopupMenuDivider(),
                    PopupMenuItem<Object>(
                      onTap: () {},
                      child: const Text('enabled'),
                    ),
                  ];
                },
                child: Container(
                  height: 30,
                  width: 30,
                  color: Colors.blue,
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

https://user-images.githubusercontent.com/32538273/234240930-296272ed-aa46-4611-9751-eb5404c27706.mov

</details>

### Logs

<details><summary>Logs</summary>

```console
Launching lib/main.dart on macOS in debug mode...
Building macOS application...                                           
Syncing files to device macOS...                                   185ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on macOS is available at: http://127.0.0.1:60107/icN9UNpDkIE=/
The Flutter DevTools debugger and profiler on macOS is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:60107/icN9UNpDkIE=/

```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[!] Flutter (Channel stable, 3.7.10, on macOS 13.3 22E252 darwin-x64, locale en-GB)
    • Flutter version 3.7.10 on channel stable at /Users/valentin/flutter/flutter
    ! Warning: `dart` on your path resolves to
      /usr/local/Cellar/dart/2.18.3/libexec/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/valentin/flutter/flutter. Consider
      adding /Users/valentin/flutter/flutter/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4b12645012 (3 weeks ago), 2023-04-03 17:46:48 -0700
    • Engine revision ec975089ac
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it
      is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /usr/local/Caskroom/android-sdk/4333796
    • Platform android-33, build-tools 30.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.3 22E252 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 112.0.5615.137

[✓] HTTP Host Availability
    • All required HTTP hosts are available
```

</details>

### PR DESCRIPTION
**Pull request by ValentinVignal**
Align the `PopupMenu` under its child.

Before:

image

After:

image

*List which issues are fixed by this PR. You must list at least one issue.*

Fixes https://github.com/flutter/flutter/issues/125474

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 1]
```diff
@@ -1238,12 +1238,19 @@ class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
...
+    late Offset offset;
     switch (popupMenuPosition) {
       case PopupMenuPosition.over:
         offset = widget.offset;
       case PopupMenuPosition.under:
-        offset = Offset(0.0, button.size.height - (widget.padding.vertical / 2)) + widget.offset;
+        offset = Offset(
+          0.0,
+          button.size.height,
+        ) + widget.offset;
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/popup_menu.dart:1249]: nit: maybe we don't have to break line since it is pretty short:)

**Review by QuncCccccc — APPROVED** [review]
Nice fix. Thanks for the contribution! LGTM:)

**Comment by QuncCccccc** [comment]
@ValentinVignal probably we need to rebase master because Google testing is failing:)

File: packages/flutter/lib/src/material/popup_menu.dart (hunk 1 above)
Comment (ValentinVignal) [reply packages/flutter/lib/src/material/popup_menu.dart]: Sure I did it in [refactor: Don't break the line](https://github.com/flutter/flutter/pull/125534/commits/3d710ec23d2d248bef4260eecf5942ddc82df827)

**Comment by ValentinVignal** [comment]
> @ValentinVignal probably we need to rebase master because Google testing is failing:)

Done :)

**Comment by QuncCccccc** [comment]
> > @ValentinVignal probably we need to rebase master because Google testing is failing:)
> 
> Done :)

Thank you! Looks like there are some conflicts in packages/flutter/test/material/popup_menu_test.dart file

**Comment by ValentinVignal** [comment]
@QuncCccccc The conflicts are fixed in
[Merge remote-tracking branch 'upstream/main' into flutter/popup-menu/…](https://github.com/flutter/flutter/pull/125534/commits/16e5a0c3810a3cd7689c331680b655d40d435473)

**Review by Piinks — APPROVED** [review]
LGTM

**Pull request merged**